src/components/interactive-table/index.tsx

```diff
@@ -94,27 +109,92 @@ const RowBackground = styled.div<{ $height: number; $width: number; $top: number
       : theme.colors.secondary + Theme.opacity.hex['005']};
 `
 
+enum SORT_DIRECTION {
+  ASC = 'asc',
+  DESC = 'desc',
+}
+
 const InteractiveTable: FC<InteractiveTableProps> = ({ columns, rows, onRowClick }) => {
+  const [sortDirection, setSortDirection] = useState<SORT_DIRECTION>(SORT_DIRECTION.ASC)
+  const [sortBy, setSortBy] = useState<string>(
+    (() => {
+      const lsVal = localStorage.getItem('odigos-sort-by')
+
+      if (!!lsVal) {
+        const found = columns.find(({ key }) => key === lsVal)
+        if (!!found) return lsVal
+      }
+
+      return 'name'
+    })()
+  )
+
+  const onSort = (key: string) => {
+    if (sortBy === key) {
+      setSortDirection((prev) => (prev === SORT_DIRECTION.ASC ? SORT_DIRECTION.DESC : SORT_DIRECTION.ASC))
+    } else {
+      setSortBy(key)
+      setSortDirection(SORT_DIRECTION.ASC)
+      localStorage.setItem('odigos-sort-by', key)
+    }
+  }
+
+  const sorted = useMemo(() => {
+    const getCellValue = (row: InteractiveTableProps['rows'][0], key: string) => {
+      return row.cells.find(({ columnKey }) => columnKey === key)?.value ?? null
+    }
+
+    const compareValues = (a: unknown, b: unknown) => {
+      if (typeof a === 'number' && typeof b === 'number') return a - b
+      if (typeof a === 'string' && typeof b === 'string') return a.localeCompare(b)
+      if (typeof a === 'boolean' && typeof b === 'boolean') return a === b ? 0 : a ? 1 : -1
+
+      // Fallback (handles null, undefined, mixed types)
+      return String(a).localeCompare(String(b))
+    }
+
+    return !!sortBy
+      ? rows.sort((a, b) => {
```

Comment:
```suggestion
      ? [...rows].sort((a, b) => {
```

I think sort is update the original array, need to double check on this, maybe it can lead to issues